include/options.h

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MIXERCLIENT_OPTIONS_H
+#define MIXERCLIENT_OPTIONS_H
+
+#include <memory>
+
+namespace google {
+namespace mixer_client {
+
+// Options controlling check behavior.
+struct CheckOptions {
+  // Default constructor.
+  CheckOptions()
+      : num_entries(10000), flush_interval_ms(500), expiration_ms(1000) {}
```

Comment:
These are based pass experience.  For example, num_entries determines how much memory will be used for cache.  flush_interval will be based on what kind of checks.  For ESP, check is only checking api_key,  flush_interval is 5 minutes. 

We could document these in the comment.